AuthenticatorSyncShared/Tests/SharedKeychainRepositoryTests.swift

```diff
@@ -0,0 +1,108 @@
+    ///
+    func testDeleteAuthenticatorKey() async throws {
+        try subject.deleteAuthenticatorKey()
+
+        let queries = try XCTUnwrap(keychainService.deleteQueries as? [[CFString: Any]])
+        XCTAssertEqual(queries.count, 1)
+
+        let query = try XCTUnwrap(queries.first)
+        try XCTAssertEqual(XCTUnwrap(query[kSecAttrAccessGroup] as? String), accessGroup)
+        try XCTAssertEqual(XCTUnwrap(query[kSecAttrAccessible] as? String),
+                           String(kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly))
+        try XCTAssertEqual(XCTUnwrap(query[kSecAttrAccount] as? String),
+                           SharedKeychainItem.authenticatorKey.unformattedKey)
+        try XCTAssertEqual(XCTUnwrap(query[kSecClass] as? String),
+                           String(kSecClassGenericPassword))
+    }
+
+    /// Verify that `getAuthenticatorKey()` returns a value successfully when one is set. Additionally, verify the
+    /// search attributes are specified correctly.
+    ///
+    func testGetAuthenticatorKey() async throws {
+        let key = SymmetricKey(size: .bits256)
+        let data = key.withUnsafeBytes { Data(Array($0)) }
+
+        keychainService.setSearchResultData(data)
+
+        let returnData = try await subject.getAuthenticatorKey()
+        XCTAssertEqual(returnData, data)
+
+        let query = try XCTUnwrap(keychainService.searchQuery as? [CFString: Any])
+        try XCTAssertEqual(XCTUnwrap(query[kSecAttrAccessGroup] as? String), accessGroup)
+        try XCTAssertEqual(XCTUnwrap(query[kSecAttrAccessible] as? String),
+                           String(kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly))
+        try XCTAssertEqual(XCTUnwrap(query[kSecAttrAccount] as? String),
+                           SharedKeychainItem.authenticatorKey.unformattedKey)
+        try XCTAssertEqual(XCTUnwrap(query[kSecClass] as? String), String(kSecClassGenericPassword))
+        try XCTAssertEqual(XCTUnwrap(query[kSecMatchLimit] as? String), String(kSecMatchLimitOne))
+        try XCTAssertTrue(XCTUnwrap(query[kSecReturnAttributes] as? Bool))
+        try XCTAssertTrue(XCTUnwrap(query[kSecReturnData] as? Bool))
+    }
+
+    /// Verify that `getAuthenticatorKey()` fails with an error when the Authenticator is not present in the keychain
+    ///
+    func testGetAuthenticatorKeyFailed() async throws {
+        let error = AuthenticatorKeychainServiceError.keyNotFound(SharedKeychainItem.authenticatorKey)
+        keychainService.searchResult = .failure(error)
+
+        do {
+            _ = try await subject.getAuthenticatorKey()
+            XCTFail("Expected error `keyNotFound` was never thrown!")
+        } catch let caughtError as AuthenticatorKeychainServiceError {
+            XCTAssertEqual(caughtError, error)
+        } catch let caughtError {
+            XCTFail(
+                "The error caught (\(caughtError)) does not match the type of error provided (\(error))."
+            )
+        }
```

Comment:
🎨 `assertAsyncThrows` can be used instead of this to be more clean leveraging the new `AuthenticatorSyncSharedTestCase`.

Reply:
This should be updated in the latest.